core/src/main/scala/cats/data/Validated.scala

```diff
@@ -1035,6 +1044,15 @@ sealed abstract private[data] class ValidatedInstances2 {
   // scalastyle:off method.length
 }
 
+private[data] class ValidatedSelective[E: Semigroup] extends ValidatedApplicative[E] with Selective[Validated[E, *]] {
+  override def select[A, B](fab: Validated[E, Either[A, B]])(ff: => Validated[E, A => B]): Validated[E, B] =
```

Comment:
is this not rigid? Looks like to me.

Reply:
It skips the effect appropriately, but is inconsistent with `ap`.

This is consistent with `ap`, but fails the skip laws, which I think could be relaxed to work only on pure `fab`.  More importantly, it fails associativity, and I'm entering the stupid hours.

```scala
  override def select[A, B](fab: Validated[E, Either[A, B]])(ff: => Validated[E, A => B]): Validated[E, B] =
    fab match {
      case Valid(Right(b)) => Valid(b)
      case Valid(Left(a)) => ff.map(_(a))
      case e @ Invalid(e1) =>
        ff match {
          case Valid(_) => e
          case Invalid(e2) => Invalid(EE.combine(e1, e2))
        }
    }
```

Reply:
It's the `Invalid - Valid(Right)) - Invalid` combination that's not associative in the above definition.  If we select `fb <*? fc` first, we get a valid function that discards the invalid `fc`, so the result is equivalent to the invalid `fa`.  If we select `fa <*? fb` first, the result is `fa`, and then combined with the invalid `fc` in the final step.

We can regain associativity by not combining two `Invalid`s, but then `ap != apS`, where `ap == apS` is the paper's definition of rigidity:

```scala
val left: F[Either[A => B, B]] = ff.map(Left(_))
val right: F[(A => B) => B] = fa.map((a: A) => _(a))
left.select(right)
```

Reply:
If you ask me, associativity is a must. The law ap == apS is one I would at least consider relaxing.

That law feels like it may preclude any rigid selects which are not also Monads.

And if you are a monad you are back to the part where we aren't really adding something new.

The struggle here feels like:

1. Add something lawful
2. Has non-trivial implementations that can't be implemented with Applicative
3. Cannot be extended to Monad.

I'm not sure if I've seen an existence proof of something matching all three.

Reply:
Several are close, but none hit the mark:

* `Const` (aka `Over`) has no `Monad` and is rigid per the paper's definition: it passes the selective laws and `ap == apS`.  It fails the our rigid skip laws.

 * There's [FreeRigidSelective](https://github.com/cb372/cats-selective/blob/master/core/src/main/scala/cats/free/FreeRigidSelective.scala).  It's `ap == apS`, but I think would also fail our rigid skip laws.  

* `Validated` and `Under` follow our skip laws, but not `ap == apS`.

*  All `FlatMap`s appear to pass `ap == apS` and selective associativity, but at least `Map[K, *]` and `Tuple[X, *]` fail our skip laws.

We may have conflated two concepts with what the paper calls rigid and our rigid skip laws, but everything that satisfies both seems to have a `Monad`.